### PR TITLE
Add new NoEmptyStyleRule with tests

### LINKED ISSUE #8: Implement "Empty Rule" lint rule
Need to implement a lint rule that reports empty rules.

```scss
.foo {
  .bar {
    color: red;
    .baz {}  // LINT
}
```

This is similar to scss-lint's [EmptyRule](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#emptyrule).

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #8

### CODE REVIEW

File: lib/src/rules/no_empty_style.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// https://github.com/sass/dart-sass/issues/236.
+import 'package:sass/src/ast/sass.dart';
+
+import '../lint.dart';
+import '../rule.dart';
+
+/// A lint rule that reports on the existence of @debug at-rules.
+///
+/// These at-rules should not be found in Sass documents, for example checked
+/// into source control.
```
Comment (nex3) [inline lib/src/rules/no_empty_style.dart]: Outdated doc comment.

File: lib/src/rules/no_empty_style.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+class NoEmptyStyleRule extends Rule {
+  NoEmptyStyleRule() : super('no_empty_style_rule');
+
+  @override
+  List<Lint> visitStyleRule(StyleRule node) {
+    var lint = <Lint>[];
+
+    // A StyleRule has style declarations if it is non-empty (`every` will
+    // always return `true` on an empty list), and if it contains anything
+    // other than these non-outputting rules).
```
Comment (nex3) [inline lib/src/rules/no_empty_style.dart]: Nit: extra closing paren.

File: lib/src/rules/no_empty_style.dart [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    var hasStyleDeclaration = !node.children.every((child) =>
+        child is DebugRule ||
+        child is ErrorRule ||
+        child is FunctionRule ||
+        child is VariableDeclaration ||
+        child is WarnRule);
```
Comment (nex3) [inline lib/src/rules/no_empty_style.dart]: Also `MixinRule` and `SilentCommentRule`.

File: lib/src/rules/no_empty_style.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/rules/no_empty_style.dart]: Done. Dunno if this "outputting" concept is the right verbiage. Any suggestions?

File: lib/src/rules/no_empty_style.dart (hunk 2 above)
Comment (srawlins) [reply lib/src/rules/no_empty_style.dart]: Done.

File: lib/src/rules/no_empty_style.dart (hunk 3 above)
Comment (srawlins) [reply lib/src/rules/no_empty_style.dart]: Done.

File: lib/src/rules/no_empty_style.dart (hunk 1 above)
Comment (nex3) [reply lib/src/rules/no_empty_style.dart]: I might say "any children that produce CSS".

**Pull request merged**